momentumopt/python/momentumopt/kinoptpy/momentum_kinematics_optimizer.py

```diff
@@ -394,23 +375,43 @@ def optimize(self, init_state, contact_sequence, dynamic_sequence, plotting=Fals
                 self.inv_kin.forward_robot(q, dq)
                 self.fill_kinematic_result(it, q, dq)
         else:
-            print("\n First order IK formulation is used, set use_second_order_inv_kin to True "
-                  "in the config file if you want to use second order IK. \n")
             for it in range(self.num_time_steps):
                 quad_goal = se3.Quaternion(se3.rpy.rpyToMatrix(np.matrix(self.base_des[:,it]).T))
                 quad_q = se3.Quaternion(float(q[6]), float(q[3]), float(q[4]), float(q[5]))
-                amom_ref = self.reg_orientation * se3.log((quad_goal * quad_q.inverse()).matrix()) + self.amom_dyn[it]
+
+                ## check if this instance belongs to the flight phase and set the momentums accordingly
+                for eff in range(len(self.eff_names)):
+                    if self.dynamic_sequence.dynamics_states[it].effActivation(eff):
+                        is_flight_phase = False
+                        break
+                    else:
+                        is_flight_phase = True
+                ## for flight phase keep the desired momentum constant
+                if is_flight_phase:
+                    lmom_ref[0:2] = mom_ref_flight[0:2]
+                    amom_ref = mom_ref_flight[3:6]
+                    lmom_ref[2] -= self.planner_setting.get(PlannerDoubleParam_RobotWeight) * self.dt
```

Comment:
Why do you need to remove the `robot weight * dt` here? Isn't the momentum in the air the last one before takeoff?

Reply:
In the air we have the force by gravity still applied to the robot, so the momentum is decayed in z-direction by this value. Otherwise the since the momentum would remain fixed, the robot would not come back to the ground :)

Reply:
Oh, yeah, I see. Makes sense :)